Quadriga/src/main/java/edu/asu/spring/quadriga/web/publicwebsite/ProjectStatsController.java

```diff
@@ -0,0 +1,119 @@
+package edu.asu.spring.quadriga.web.publicwebsite;
+
+import java.util.List;
+
+import javax.xml.bind.JAXBException;
+
+import org.codehaus.jettison.json.JSONArray;
+import org.codehaus.jettison.json.JSONException;
+import org.codehaus.jettison.json.JSONObject;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.PropertySource;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import edu.asu.spring.quadriga.domain.IConceptStats;
+import edu.asu.spring.quadriga.domain.network.INetwork;
+import edu.asu.spring.quadriga.domain.workbench.IProject;
+import edu.asu.spring.quadriga.exceptions.QuadrigaStorageException;
+import edu.asu.spring.quadriga.service.network.INetworkManager;
+import edu.asu.spring.quadriga.service.publicwebsite.impl.ProjectStats;
+import edu.asu.spring.quadriga.service.workbench.IRetrieveProjectManager;
+
+/**
+ * This controller has all the mappings required to view the statistics of the
+ * project
+ * 
+ * @author ajaymodi
+ *
+ */
+
+@PropertySource(value = "classpath:/user.properties")
+@Controller
+public class ProjectStatsController {
+
+    @Autowired
+    private IRetrieveProjectManager projectManager;
+
+    @Autowired
+    private INetworkManager networkmanager;
+
+    @Autowired
+    private ProjectStats projectStats;
+
+    @Autowired
+    private Environment env;
+
+    private JSONArray getProjectStatsJson(List<IConceptStats> topConcepts)
+            throws JSONException {
+        JSONArray jsonArray = new JSONArray();
+
+        int cnt = Integer.parseInt(env.getProperty("project.stats.topcount"));
+        int len = topConcepts.size() > cnt ? cnt : topConcepts.size();
+
+        for (int i = 0; i < len; i++) {

```

Comment:
I wouldn't do that here, but call the method with only the concepts that you want in JSON. You can simply do a subList call on the original list before passing it to this method.